packages/schema-extract/README.md

```diff
@@ -0,0 +1,217 @@
+# Schema Extract
+
+
+Schema extract transforms code files into [JSON-Schema](https://json-schema.org/) schemas. Schema extraction is a necessary step to create automated documentation for your code.
+
+It is composed of two parts, the **ts transformer** and **linker**.
+
+
+## Playground
+The `schema-extract` package has a playground that allows live editing of your code and transforms it to a schema and displays the results as you type it.
+
+## Usage
+
+Import the `createLinker` function from `@ui-autotools/schema-extract`. This function receives a string array of file paths and returns an initalized linker class.
+To get the linked schema of a specific file invoke `linker.flatten(fileName)`:
+
+```
+    import {createLinker} from '@ui-autotools/schema-extract';
+    const files = [
+        'myProject/src/file-a.ts',
+        'myProject/src/file-b.ts'
+    ];
+    const linker = createLinker(files);
+    const linkedSchema = linker.flatten(files[0], 'IAnimal');
+
+```
+
+## Example
+
+Take a look at the following interface:
+```
+// file-a.ts
+    export interface IAnimal {
+        name: string;
+        hasTail: boolean;
+        makeNoise: (sound: ISound) => void;
+        isTamed?: 'YES' | 'NO';
+    }
+```
+If we run the command from the usage section it will be transformed to:
+```
+{
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "$id": "/index.tsx",
+  "$ref": "common/module",
+  "properties": {},
+  "definitions": {
+    "IAnimal": {
+      "properties": {
+        "name": {
+          "type": "string"
+        },
+        "hasTail": {
+          "type": "boolean"
+        },
+        "makeNoise": {
+          "$ref": "common/function",
+          "arguments": [
+            {
+              "$ref": "#ISound",
+              "name": "sound"
+            }
+          ],
+          "returns": {
+            "$ref": "common/undefined"
+          },
+          "requiredArguments": [
+            "sound"
+          ]
+        },
+        "isTamed": {
+          "type": "string",
+          "enum": [
+            "YES",
+            "NO"
+          ]
+        }
+      },
+      "required": [
+        "name",
+        "hasTail",
+        "makeNoise"
+      ],
+      "$ref": "common/interface"
+    }
+  }
+}
+```
+
+## TS Transformer
+
+Transforms a typescript source file into JSON-Schema. 
```

Comment:
mention that it is being used by default when using createLinker. No need to use this specifically unless requiring non-linked schema